src/ExperimentalStableMemory.mo

```diff
@@ -41,58 +46,302 @@ module {
   /// Initially `0`.
   /// Preserved across upgrades, together with contents of allocated
   /// stable memory.
+  ///
+  /// Example:
+  /// ```motoko no-repl
+  /// let beforeSize = StableMemory.size();
+  /// ignore StableMemory.grow(10);
+  /// let afterSize = StableMemory.size();
+  /// afterSize - beforeSize // => 10
+  /// ```
   public let size : () -> (pages : Nat64) = Prim.stableMemorySize;
 
-  /// Grow current `size` of stable memory by `pagecount` pages.
+  /// Grow current `size` of stable memory by the given number of pages.
   /// Each page is 64KiB (65536 bytes).
-  /// Returns previous `size` when able to grow.
+  /// Returns the previous `size` when able to grow.
   /// Returns `0xFFFF_FFFF_FFFF_FFFF` if remaining pages insufficient.
-  /// Every new page is zero-initialized, containing byte 0 at every offset.
+  /// Every new page is zero-initialized, containing byte 0x00 at every offset.
   /// Function `grow` is capped by a soft limit on `size` controlled by compile-time flag
   ///  `--max-stable-pages <n>` (the default is 65536, or 4GiB).
-  public let grow : (new_pages : Nat64) -> (oldpages : Nat64) = Prim.stableMemoryGrow;
+  ///
+  /// Example:
+  /// ```motoko no-repl
+  /// import Error "mo:base/Error";
+  ///
+  /// let beforeSize = StableMemory.grow(10);
+  /// if (beforeSize == 0xFFFF_FFFF_FFFF_FFFF) {
+  ///   throw Error.reject("Out of memory");
+  /// };
+  /// let afterSize = StableMemory.size();
+  /// afterSize - beforeSize // => 10
+  /// ```
+  public let grow : (newPages : Nat64) -> (oldPages : Nat64) = Prim.stableMemoryGrow;
 
   /// Returns a query that, when called, returns the number of bytes of (real) IC stable memory that would be
   /// occupied by persisting its current stable variables before an upgrade.
   /// This function may be used to monitor or limit real stable memory usage.
   /// The query computes the estimate by running the first half of an upgrade, including any `preupgrade` system method.
   /// Like any other query, its state changes are discarded so no actual upgrade (or other state change) takes place.
   /// The query can only be called by the enclosing actor and will trap for other callers.
+  ///
+  /// Example:
+  /// ```motoko no-repl
+  /// actor {
+  ///   public func example() : async Text {
+  ///     let memoryUsage = StableMemory.stableVarQuery();
+  ///     let beforeSize = (await memoryUsage()).size;
+  ///     ignore StableMemory.grow(10);
+  ///     let afterSize = (await memoryUsage()).size;
+  ///     debug_show (afterSize - beforeSize)
+  ///   };
+  /// };
```

Comment:
```suggestion
  /// actor {
  ///   stable var state = "";
  ///   public func example() : async Text {
  ///     let memoryUsage = StableMemory.stableVarQuery();
  ///     let beforeSize = (await memoryUsage()).size;
  ///     state #= "abcdefghijklmnopqrstuvwxyz";
  ///     let afterSize = (await memoryUsage()).size;
  ///     debug_show (afterSize - beforeSize)
  ///   };
  /// };
```

Your example will always return a diff of 0 since stableVarQuery give you the size of stable variables (in serialized form), not user allocated pages.
This example might be better (always returns a diff of 26).

https://m7sm4-2iaaa-aaaab-qabra-cai.raw.ic0.app/?tag=3043446797

Reply:
Much better; thanks. The new example also seems like a much more realistic use case.